File: abc/abc209/abc209_c.fs
// https://atcoder.jp/contests/abc209/submissions/24165053

let MOD = 1000000007L
let modmul x y = ((x % MOD) * (y % MOD)) % MOD

stdin.ReadLine()
stdin.ReadLine().Split() |> Array.map int64
|> Array.sort
|> Array.mapi(fun i x -> max 0L (x - int64 i))
|> Array.fold(fun s x -> modmul s x) 1L
|> stdout.WriteLine

File: abc/abc209/abc209_c.cs
// https://atcoder.jp/contests/abc209/submissions/24162906

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var N = long.Parse(Console.ReadLine());
        var C = Console.ReadLine().Split().Select(long.Parse).ToArray();
        Array.Sort(C);

        var result = 1L;
        for (int i = 0; i < N; i++)
        {
            result *= Math.Max(0, C[i] - i);
            result %= 1000000007L;
        }
        Console.WriteLine(result);
    }
}
